File: extract.js
import fs from 'fs';

const transcriptPath = 'C:\\Users\\shubham\\.gemini\\antigravity-ide\\brain\\35f6e5a3-abdc-4199-b5ae-019836311733\\.system_generated\\logs\\transcript.jsonl';
const targetFilePath = 'C:\\Users\\shubham\\Downloads\\gitgithubnotes\\web-app\\src\\Terraform_Complete_Notes.md';

const content = fs.readFileSync(transcriptPath, 'utf-8');
const lines = content.split('\n');

let extractedText = '';

for (const line of lines) {
    if (!line) continue;
    try {
        const data = JSON.parse(line);
        if (data.type === 'TOOL_RESPONSE' && data.tool_calls) {
            for (const call of data.tool_calls) {
                if (call.function_name === 'default_api:view_file' && call.response && call.response.output) {
                    const output = call.response.output;
                    if (output.includes('==Start of OCR')) {
                        // Extract text using regex
                        const regex = /==Start of OCR for page \d+==\r?\n([\s\S]*?)\r?\n==End of OCR for page \d+==/g;
                        let match;
                        while ((match = regex.exec(output)) !== null) {
                            let pageText = match[1];
                            
                            // Basic heading formatting
                            pageText = pageText.replace(/^(\d+\.\s+[A-Z].+)$/gm, '## $1');
                            pageText = pageText.replace(/^([a-z]\.\s+[A-Z].+)$/gm, '### $1');
                            
                            // Fix some spacing issues from OCR
                            pageText = pageText.replace(/\r\n/g, '\n');
                            
                            extractedText += pageText + '\n\n';
                        }
                    }
                }
            }
        }
    } catch (e) {
        // ignore JSON parse errors on malformed lines
    }
}

if (extractedText) {
    extractedText = '\n\n# --- EXTRACTED PDF CONTENT (Full Depth) ---\n\n' + extractedText;
    fs.appendFileSync(targetFilePath, extractedText);
    console.log('Appended ' + extractedText.length + ' characters to ' + targetFilePath);
} else {
    console.log('No OCR text found. Attempting fallback text search...');
}

File: extract.py
import fitz  # PyMuPDF
import re
import sys
import os

pdf_path = r"C:\Users\shubham\Downloads\gitgithubnotes\Terraform Notes Complete.pdf"
target_path = r"C:\Users\shubham\Downloads\gitgithubnotes\web-app\src\Terraform_Complete_Notes.md"

def extract_pdf():
    print(f"Opening {pdf_path}...")
    try:
        doc = fitz.open(pdf_path)
    except Exception as e:
        print(f"Error opening PDF: {e}")
        sys.exit(1)

    extracted_text = "\n\n# --- EXTRACTED PDF CONTENT (Complete Depth) ---\n\n"
    
    print("Extracting pages...")
    for page_num, page in enumerate(doc):
        text = page.get_text()
        
        # Apply some basic markdown formatting heuristics
        # Make things that look like headers actual markdown headers
        lines = text.split('\n')
        formatted_lines = []
        for line in lines:
            stripped = line.strip()
            # If line starts with a number followed by dot (e.g. "1. Introduction")
            if re.match(r'^\d+\.\s+[A-Z]', stripped):
                formatted_lines.append(f"\n## {stripped}\n")
            # If line starts with a letter followed by dot (e.g. "a. What is...")
            elif re.match(r'^[a-z]\.\s+[A-Z]', stripped):
                formatted_lines.append(f"\n### {stripped}\n")
            else:
                formatted_lines.append(stripped)
        
        page_text = "\n".join(formatted_lines)
        
        extracted_text += page_text + "\n\n"
        
    print(f"Extracted {len(extracted_text)} characters.")
    
    print(f"Appending to {target_path}...")
    with open(target_path, "a", encoding="utf-8") as f:
        f.write(extracted_text)
        
    print("Done!")

if __name__ == "__main__":
    extract_pdf()
